repos/spack_repo/builtin/packages/octave_statistics/package.py

```diff
@@ -12,9 +12,13 @@ class OctaveStatistics(OctavePackage, SourceforgePackage):
     """Additional statistics functions for Octave."""
 
     homepage = "https://octave.sourceforge.io/statistics/"
-    sourceforge_mirror_path = "octave/statistics-1.4.2.tar.gz"
+    # sourceforge_mirror_path = "octave/statistics-1.4.2.tar.gz"
+    git = "https://github.com/gnu-octave/statistics/"
+    url = "https://github.com/gnu-octave/statistics/releases/download/release-1.7.6/statistics-1.7.6.tar.gz"
 
+    version("1.7.6", sha256="a518c50209e25e59742414c73955060f83c39c07b2d9a20b2fc8d13bd3106af3")
```

Comment:
Did you check if the has for v1.4.2 needs changes?

Reply:
Hop, no, indeed 1.4.2 is not on github. Alike https://github.com/spack/spack-packages/pull/2244 it fails.
It seems that the packages that no longer update sourceforge files do not put old versions on github.
And I failed to make a `url_for_version` function that suits both.

Reply:
It seems the def url_for_version fails to collaborate with spack checksum.
Here is another example:
```
root@3ab6b6d3802c:/# spack checksum parallel-netcdf@1.9.0
==> Selected 0 of 11 versions. Filtered by @1.9.0

==> Enter number of versions to take, or use a command:
    [c]hecksum  [e]dit  [f]ilter  [a]sk each  [n]ew only  [r]estart  [q]uit
action> c
root@3ab6b6d3802c:/# 
```
What do you think @alalazo ?